ykrt/src/compile/jitc_yk/jit_ir/mod.rs

```diff
@@ -1,9 +1,45 @@
 //! The JIT's Intermediate Representation (IR).
 //!
-//! This is the IR created by [trace_builder] and which is then optimised. The IR can feel a little
-//! odd at first because we store indexes into vectors rather than use direct references. This
-//! allows us to squeeze the amount of the memory used down (and also bypasses issues with
-//! representing graph structures in Rust, but that's slightly accidental).
+//! This is the IR created by [trace_builder] and which is then analysed and optimised. The IR has
+//! the following properties:
+//!
+//!   1. We use vectors rather than direct references. This allows us to squeeze the amount of the
+//!      memory used down (and also bypasses issues with representing graph structures in Rust, but
+//!      that's slightly accidental).
+//!
+//!   2. The IR is designed to allow most manipulations to be performed without requiring
+//!      reallocation.
+//!
+//! This second property has some surprising consequences:
+//!
+//!   1. We have to introduce `Proxy*` instructions which semi-transparently "forward" an
+//!      instruction's results on. For example if we IR in memory stored along these lines:
```

Comment:
"results on"? Also "if we HAVE IR in memory".

Reply:
Fixed in 85ef34ee.